generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/execution/FlowSubscriptionExecutionStrategy.kt

```diff
@@ -103,21 +104,33 @@ class FlowSubscriptionExecutionStrategy(dfe: DataFetcherExceptionHandler) : Exec
             Let {fieldStream} be the result of running {ResolveFieldEventStream(subscriptionType, initialValue, rootField, argumentValues)}.
             Return {fieldStream}.
      */
+    @Suppress("UNCHECKED_CAST")
     private fun createSourceEventStream(
         executionContext: ExecutionContext,
         parameters: ExecutionStrategyParameters
     ): CompletableFuture<Flow<*>?> {
         val newParameters = firstFieldOfSubscriptionSelection(parameters)
 
         val fieldFetched = fetchField(executionContext, newParameters)
-        return fieldFetched.thenApply { fetchedValue ->
+        return if (fieldFetched is CompletableFuture<*>) {
+            val promiseFieldFetched = fieldFetched as CompletableFuture<FetchedValue>
+            promiseFieldFetched.thenApply { fetchedValue ->
+                when (val publisherOrFlow: Any? = fetchedValue.fetchedValue) {
+                    is Publisher<*> -> publisherOrFlow.asFlow()
+                    // below explicit cast is required due to the type erasure and Kotlin declaration-site variance vs Java use-site variance
+                    is Flow<*> -> publisherOrFlow
+                    else -> null
+                }
+            }
+        } else {
+            val fetchedValue = fieldFetched as FetchedValue
             val flow = when (val publisherOrFlow: Any? = fetchedValue.fetchedValue) {
                 is Publisher<*> -> publisherOrFlow.asFlow()
                 // below explicit cast is required due to the type erasure and Kotlin declaration-site variance vs Java use-site variance
                 is Flow<*> -> publisherOrFlow
                 else -> null
             }
-            flow
+            CompletableFuture.completedFuture(flow)
```

Comment:
I dont' think this will work -> this will mark the execution as complete with the results from completing the flows so it is not really streaming response back anymore

Reply:
should we just imply that the result of `fetchField` will always be a `CompletableFuture` ?

Reply:
just commited the change to always wrap.